marko/element.py

```diff
@@ -0,0 +1,24 @@
+from .helpers import camel_to_snake_case, is_type_check
+
+if is_type_check():
+    from typing import Any
+
+class Element:
+    """This class holds attributes common to both the BlockElement and 
+    InlineElement classes.
+    This class should not be subclassed by any other classes beside these.
+    """
+    @classmethod
+    def get_element_type(cls, snake_case=False):  # type: (Any) -> str
+        """
+        Return the Markdown element type that the object represents.
+
+        :param snake_case: Return the element type name in snake case if True
+        """
+
+        # Prevent override of BlockElement and InlineElement
+        if cls.override and cls.__base__ not in Element.__subclasses__():
```

Comment:
```suggestion
        if cls.override and cls.__base__ in Element.__subclasses__():
```

Should be `in`?

Reply:
Is multi-inheritance possible here?

Reply:
I'm still pretty new to the code, but I think there are two scenarios where override is True:

1. subclassing a 'normal' element, such as a `Heading`
2. subclassing `BlockElement` or `InlineElement`

In 1, the user's intention is to clobber `Heading`, and so the parent class name is returned.
In 2, it seems to me that an override would be a mistake - a user shouldn't be able to override `BlockElement` or `InlineElement`.

So to allow an override, we have to check that they're **not** trying to override `BlockElement` or `InlineElement`, the subclasses of `Element`. In that case, their custom class name is returned.

I guess multi-inheritance wouldn't work, as  `__base__` will only take the first element of `__bases__`

Reply:
Fine, I think multi-inheritance is not likely to happen though.